modules/renter/download.go

```diff
@@ -116,35 +121,58 @@ type (
 	}
 )
 
-// newDownload initializes and returns a download object.
-func (r *Renter) newDownload(f *file, destination string, currentContracts map[modules.NetAddress]types.FileContractID) *download {
-	d := &download{
-		finishedChunks: make([]bool, f.numChunks()),
+// newSectionDownload initialises and returns a download object for the specified chunk.
+func (r *Renter) newSectionDownload(f *file, destination modules.DownloadWriter, currentContracts map[modules.NetAddress]types.FileContractID, offset, length uint64) *download {
+	d := &download{}
+	d.initDownload(f, destination)
 
-		startTime: time.Now(),
+	// Settings specific to a chunk download.
+	d.offset = offset
+	d.length = length
 
-		chunkSize:   f.chunkSize(),
-		destination: destination,
-		erasureCode: f.erasureCode,
-		fileSize:    f.size,
-		masterKey:   f.masterKey,
-		numChunks:   f.numChunks(),
-		siapath:     f.name,
+	// Calculate chunks to download.
+	min_chunk := uint64(math.Floor(float64(offset / f.chunkSize())))
+	max_chunk := uint64(math.Floor(float64((offset + length) / f.chunkSize())))
 
-		downloadFinished: make(chan struct{}),
-	}
+	makeRange(int(min_chunk), int(max_chunk+1), &d.finishedChunks)
```

Comment:
this function call should be inlined. It's only used once and its effect isn't immediately obvious.